recipes-kernel/linux/smarc-rzg2l/0001-add-vsc8531-userspace-dts.patch

```diff
@@ -1,18 +1,16 @@
 diff --git a/r9a07g044l2-smarc.dts.orig b/r9a07g044l2-smarc.dts
-index 651aa6b..f165a7a 100644
+index bc2af6c..82ba67e 100644
 --- a/arch/arm64/boot/dts/renesas/r9a07g044l2-smarc.dts
 +++ b/arch/arm64/boot/dts/renesas/r9a07g044l2-smarc.dts
-@@ -11,6 +11,7 @@
+@@ -11,8 +11,32 @@
  #include "rzg2l-smarc-pinfunction.dtsi"
  #include "rz-smarc-common.dtsi"
  #include "rzg2l-smarc.dtsi"
 +#include <dt-bindings/net/mscc-phy-vsc8531.h>
  
  / {
  	model = "Renesas SMARC EVK based on r9a07g044l2";
-@@ -46,3 +47,25 @@
- 	/delete-node/ sd0-dev-sel-hog;
- 	/delete-node/ sd1-pwr-en-hog;
```

Comment:
The pins for P41_1 (GPIO_SD0_DEV_SEL) and P39_2 (SDIO_PWR_EN) are used on Sharc only, not on MistySOM.  
By removing sd0-dev-sel-hog and sd1-pwr-en-hog from the .dts tree for Smarc, these pins now become available for GPIO usage.  If the pins have the gpio-hog attribute or property (see the original .dts code in rzg2l-smarc-som.dtsi below), then they are exclusively owned by the compatible driver, and no other user space applications can use those pins.
        /*
	 * SD0 device selection is XOR between GPIO_SD0_DEV_SEL and SW1[2]
	 * The below switch logic can be used to select the device between
	 * eMMC and microSD, after setting GPIO_SD0_DEV_SEL to high in DT.
	 * SW1[2] should be at position 2/OFF to enable 64 GB eMMC
	 * SW1[2] should be at position 3/ON to enable uSD card CN3
	 */
	sd0-dev-sel-hog {
		**gpio-hog;**
		gpios = <RZG2L_GPIO(41, 1) GPIO_ACTIVE_HIGH>;
		output-high;
		line-name = "sd0_dev_sel";
	};

Reply:
P.S.  the sd0-dev-sel-hog and sd1-pwr-en-hog nodes are child nodes of the labelled pinctrl node in r9a07g044.dtsi file which is subsequently used by the &pinctrl reference.
```
		pinctrl: pin-controller@11030000                                               {
			compatible = "renesas,r9a07g044-pinctrl";
			reg = <0 0x11030000 0 0x10000>, <0 0x110a0020 0 0x30>;
			gpio-controller;
			#gpio-cells = <2>;
			gpio-ranges = <&pinctrl 0 0 392>;
			clocks = <&cpg CPG_MOD R9A07G044_GPIO_HCLK>;
			power-domains = <&cpg>;
			resets = <&cpg R9A07G044_GPIO_RSTN>,
				 <&cpg R9A07G044_GPIO_PORT_RESETN>,
				 <&cpg R9A07G044_GPIO_SPARE_RESETN>;
			interrupt-controller;
			#interrupt-cells = <2>;
			interrupts =	<GIC_SPI 444 IRQ_TYPE_LEVEL_HIGH>,
					<GIC_SPI 445 IRQ_TYPE_LEVEL_HIGH>,
					<GIC_SPI 446 IRQ_TYPE_LEVEL_HIGH>,
					<GIC_SPI 447 IRQ_TYPE_LEVEL_HIGH>,
```

Reply:
In the .dtsi file, r9a07g044-pinctrl is the compatible driver which would otherwise own the sd0-dev-sel-hog and sd1-pwr-en-hog nodes pins exclusively if they were not otherwise deleted.